flight/Libraries/math/smoothcontrol.c

```diff
@@ -0,0 +1,217 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <math.h>
+
+#include "openpilot.h"
+#include "pios.h"
+#include "smoothcontrol.h"
+
+#define ASSERT(x) if(x) PIOS_Assert(0);
+#define MAX(v,x) ((v) > (x) ? (x) : (v))
```

Comment:
We should probably have a good min/max implementation in pios.h or openpilot.h rather than random ones like this. e.g. http://stackoverflow.com/a/3437484. But I thought arm-none-eabi headers already give us MIN/MAX??